### ISSUE TITLE
Implement different activation functions

### ISSUE THREAD

**Issue by sakkurax7** [sakkurax7/NMIST-NeuralNetwork#1]
Specifically softmax in order to actually get backprop to work on the MNIST dataset.
